invenio_users_resources/services/groups/service.py

```diff
@@ -23,23 +32,128 @@
 class GroupsService(RecordService):
     """User groups service."""
 
+    @unit_of_work()
+    def create(self, identity, data, raise_errors=True, uow=None):
+        """Create a new group/role."""
+        self.require_permission(identity, "create")
+        data, errors = self.schema.load(
+            data or {},
+            context={"identity": identity},
+            raise_errors=raise_errors,
+        )
+
+        # Prepare role data for creation
+        name = data.get("name")
+        role_kwargs = {
+            "id": name,
+            "name": name,
+        }
+        if "description" in data:
+            role_kwargs["description"] = data["description"]
+
+        # Create group using API
+        group = GroupAggregate.create(role_kwargs)
+
+        current_app.logger.info(f"Group created: '{group.name}' by user {identity.id}")
+
+        self.run_components(
+            "create",
+            identity,
+            data=data,
+            group=group,
+            errors=errors,
+            uow=uow,
+        )
+
+        uow.register(RecordCommitOp(group, indexer=self.indexer, index_refresh=True))
+        return self.result_item(
+            self, identity, group, errors=errors, links_tpl=self.links_item_tpl
+        )
+
     def read(self, identity, id_):
         """Retrieve a user group."""
-        # resolve and require permission
         group = GroupAggregate.get_record_by_name(id_)
         if group is None:
-            # return 403 even on empty resource due to security implications
             raise PermissionDeniedError()
-
         self.require_permission(identity, "read", record=group)
-
-        # run components
         for component in self.components:
             if hasattr(component, "read"):
                 component.read(identity, group=group)
-
         return self.result_item(self, identity, group, links_tpl=self.links_item_tpl)
 
+    @unit_of_work()
+    def update(self, identity, id_, data, raise_errors=True, uow=None):
+        """Update an existing group."""
+        group = GroupAggregate.get_record_by_name(id_)
+        if group is None:
+            raise PermissionDeniedError()
+        self.require_permission(identity, "update", record=group)
+
+        data, errors = self.schema.load(
+            data or {},
+            schema_args={"partial": True},
+            context={"identity": identity, "record": group},
+            raise_errors=raise_errors,
+        )
+
+        # Business logic: validate name change
+        if "name" in data and data["name"] != group.name:
+            raise ValidationError({"name": [_("Renaming roles is not supported.")]})
+
+        # Update group using API
+        group.update(data)
```

Comment:
I think we typically place the update in a component, but I don't have strong feelings about it at this point.